dynamic-ssl/src/main/java/org/wildfly/security/dynamic/ssl/DynamicSSLContextSpiImpl.java

```diff
@@ -0,0 +1,146 @@
+            throw new IllegalStateException(ElytronMessages.log.cannotObtainConfiguredDefaultSSLContext());
+        }
+    }
+
+    @Override
+    protected void engineInit(KeyManager[] keyManagers, TrustManager[] trustManagers, SecureRandom secureRandom) {
+        // ignore
+    }
+
+    @Override
+    protected SSLSocketFactory engineGetSocketFactory() {
+        if (dynamicSSLContextImpl == null) {
+                return this.getConfiguredDefaultSSLContext().getSocketFactory();
+        }
+        if (sslSocketFactory == null) {
+            synchronized (this) {
+                if (sslSocketFactory == null) {
+                    sslSocketFactory = new DynamicSslSocketFactory(this.getConfiguredDefaultSSLContext().getSocketFactory(), dynamicSSLContextImpl);
+                }
+            }
+        }
+        return sslSocketFactory;
+    }
+
+    @Override
+    protected SSLServerSocketFactory engineGetServerSocketFactory() {
+        return this.getConfiguredDefaultSSLContext().getServerSocketFactory();
+    }
+
+    @Override
+    protected SSLEngine engineCreateSSLEngine() {
+            return this.getConfiguredDefaultSSLContext().createSSLEngine();
+    }
+
+    @Override
+    protected SSLEngine engineCreateSSLEngine(String host, int port) throws IllegalStateException {
+        try {
+            if (dynamicSSLContextImpl == null) {
+                return this.getConfiguredDefaultSSLContext().createSSLEngine(host, port);
+            }
+            SSLContext sslContext = dynamicSSLContextImpl
+                    .getSSLContext(new URI(null, null, host, port, null, null, null));
+            if (sslContext == null) {
+                throw new IllegalStateException(ElytronMessages.log.receivedSSLContextFromDynamicSSLContextProviderWasNull());
+            }
+            if (sslContext instanceof DynamicSSLContext && sslContext.getSocketFactory().equals(this.engineGetSocketFactory())) {
+                throw new IllegalStateException(ElytronMessages.log.dynamicSSLContextCreatesLoop());
+            }
+            return sslContext.createSSLEngine(host, port);
+        } catch (URISyntaxException e) {
+            throw new IllegalStateException(ElytronMessages.log.couldNotCreateURI());
+        } catch (DynamicSSLContextException e) {
+            throw new IllegalStateException(ElytronMessages.log.couldNotCreateDynamicSSLContextEngine());
+        }
+    }
+
+    @Override
+    protected SSLSessionContext engineGetServerSessionContext() {
+        throw new UnsupportedOperationException(ElytronMessages.log.dynamicSSLContextDoesNotSupportSessions());
+    }
+
+    @Override
+    protected SSLSessionContext engineGetClientSessionContext() {
+        throw new UnsupportedOperationException(ElytronMessages.log.dynamicSSLContextDoesNotSupportSessions());
+
+    }
+
+    @Override
+    protected SSLParameters engineGetSupportedSSLParameters() {
+            return this.getConfiguredDefaultSSLContext().getSupportedSSLParameters();
```

Comment:
Very minor but looks like the spacing is off here.

Reply:
Fixed